### PR TITLE
Custom video count

### PR DESCRIPTION
**Pull request by SukanyaSen530**
- Increment view count as soon as user visits single video page
- Added the changes in controller
- Updated views in liked, history and watchlater pages

### CODE REVIEW

**Review by biswanath-tewari — APPROVED** [review]

**Review by biswanath-tewari — COMMENTED** [review]
the code looks clean

File: src/context/reducers/dataReducer.js [hunk 1]
```diff
@@ -5,6 +5,12 @@ import {
   historyConstants,
   playlistConstants,
 } from "../actions/dataActions";
+import { tokenName } from "../providers/AuthProvider";
+
+const isLoggedIn = () => {
+  if (window.localStorage.getItem(tokenName)) return true;
+  else return false;
```
Comment (JowelTisso) [inline src/context/reducers/dataReducer.js:12]: Is accessing the localStorage through window necessary

File: src/context/reducers/dataReducer.js [hunk 2]
```diff
@@ -30,15 +36,59 @@ const userReducer = (state, action) => {
...
+          },
+          // playlist: {
+          //   ...state.playlist,
+          //   items: state.playlist.items.map((item) =>
+          //     item.videos.map((video) =>
+          //       video._id === payload._id ? payload : video
+          //     )
+          //   ),
+          // },
+          vid: {
```
Comment (JowelTisso) [inline src/context/reducers/dataReducer.js]: * Unused code

File: src/context/reducers/dataReducer.js (hunk 1 above)
Comment (SukanyaSen530) [reply src/context/reducers/dataReducer.js:12]: no, can be omitted if there is no other variable with same name.

File: src/pages/VideoPage/VideoPage.jsx [hunk 3]
```diff
@@ -153,6 +161,20 @@ const Video = () => {
...
+          <p className="video-section__info">
+            <span>
+              {" "}
+              Aired on{" "}
+              {new Date(createdAt).toLocaleDateString("en-US", {
+                weekday: "long",
+                year: "numeric",
+                month: "long",
+                day: "numeric",
+              })}
```
Comment (JowelTisso) [inline src/pages/VideoPage/VideoPage.jsx]: * You can extract this out

**Review by JowelTisso — APPROVED** [review]
* Rest looks good
* Clean code

File: src/context/reducers/dataReducer.js (hunk 1 above)
Comment (JowelTisso) [reply src/context/reducers/dataReducer.js:12]: You have another variable named localStorage?

File: src/pages/VideoPage/VideoPage.jsx (hunk 3 above)
Comment (SukanyaSen530) [reply src/pages/VideoPage/VideoPage.jsx]: fixed - [9c439eb](https://github.com/SukanyaSen530/DrawFlix/pull/13/commits/9c439eba9ce6c798f6ff67915789c3938e1afc6d)

File: src/context/reducers/dataReducer.js (hunk 2 above)
Comment (SukanyaSen530) [reply src/context/reducers/dataReducer.js]: removed - [9c439eb](https://github.com/SukanyaSen530/DrawFlix/pull/13/commits/9c439eba9ce6c798f6ff67915789c3938e1afc6d)

File: src/context/reducers/dataReducer.js (hunk 1 above)
Comment (SukanyaSen530) [reply src/context/reducers/dataReducer.js:12]: no, it can be removed 👍🏼

**Pull request merged**